csharp_dotnetcore/14.NLP-With-Dispatch/Startup.cs

```diff
@@ -0,0 +1,255 @@
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.ApplicationInsights;
+using Microsoft.ApplicationInsights.Extensibility;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Bot.Builder.AI.Luis;
+using Microsoft.Bot.Builder.AI.QnA;
+using Microsoft.Bot.Builder.BotFramework;
+using Microsoft.Bot.Builder.Integration.AspNet.Core;
+using Microsoft.Bot.Configuration;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using NLP_With_Dispatch_Bot.AppInsights;
+
+namespace NLP_With_Dispatch_Bot
+{
+    /// <summary>
+    /// The Startup class configures services and the app's request pipeline.
+    /// </summary>
+    public class Startup
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Startup"/> class.
+        /// This method gets called by the runtime. Use this method to add services to the container.
+        /// For more information on how to configure your application, visit https://go.microsoft.com/fwlink/?LinkID=398940.
+        /// </summary>
+        /// <param name="env">Provides information about the web hosting environment an application is running in.</param>
+        /// <seealso cref="https://docs.microsoft.com/en-us/aspnet/core/fundamentals/startup?view=aspnetcore-2.1"/>
+        public Startup(IHostingEnvironment env)
+        {
+            var builder = new ConfigurationBuilder()
+                .SetBasePath(env.ContentRootPath)
+                .AddJsonFile("appsettings.json", optional: true, reloadOnChange: true)
+                .AddJsonFile($"appsettings.{env.EnvironmentName}.json", optional: true)
+                .AddEnvironmentVariables();
+
+            Configuration = builder.Build();
+        }
+
+        /// <summary>
+        /// Gets the configuration that represents a set of key/value application configuration properties.
+        /// </summary>
+        /// <value>
+        /// The <see cref="IConfiguration"/> that represents a set of key/value application configuration properties.
+        /// </value>
+        public IConfiguration Configuration { get; }
+
+        /// <summary>
+        /// This method gets called by the runtime. Use this method to add services to the container.
+        /// </summary>
+        /// <param name="services">Specifies the contract for a collection of service descriptors.</param>
+        /// <seealso cref="IServiceCollection"/>
+        /// <seealso cref="https://docs.microsoft.com/en-us/aspnet/web-api/overview/advanced/dependency-injection"/>
+        public void ConfigureServices(IServiceCollection services)
+        {
+            // Load the connected services from .bot file
+            // BUGBUG: This needs to change to common bot file, once emulator understand appInsights type.
+            var botConfig = BotConfiguration.Load(@".\NLP-With-Dispatch-Bot.bot");
```

Comment:
This should be read from appsettings.json. The bot file will typically be encrypted once you have keys in them. The key itself and the bot file path should be in appsettings.json.

Reply:
Good point! This is the way we are doing it in the checked in samples. We should discuss in standup and move them all to this (or none) so they are consistent.

https://github.com/Microsoft/BotBuilder-Samples/blob/v4/csharp_dotnetcore/11.QnAMaker/Startup.cs